CRM/Utils/Check/Component/Env.php

```diff
@@ -939,6 +939,8 @@ public function checkMysqlUtf8mb4() {
       return $messages;
     }
 
+    // Force utf8mb4 query to throw exception as the check expects.
+    $errorScope = CRM_Core_TemporaryErrorScope::useException();
```

Comment:
Yes this fixes the issue.  One thing I don't understand is why we have to assign the result to a variable ie. `$errorScope`.  Without that assign it seems like the call to `CRM_Core_TemporaryErrorScope::useException();` has no effect and I still get a fatal error.  @totten/@seamuslee001 do you have any ideas why?

Reply:
When the variable goes out of scope, its destructor method reverts the error handling.  If there is no reference to the object then presumably the destructor is called immediately.

I don't know why all of this is needed - seems preferable to always throw/catch exceptions so this kind of thing is predictable. I guess some legacy code relies on the fatal errors?

Reply:
@mfb more likely that no body has ever gone back to this https://issues.civicrm.org/jira/browse/CRM-11043 i think we are now slowly in the process of doing what Tim Suggested in point 4 which is to deprecate CRM_Core_Error::fatal()

Reply:
I guess what must've tripped me up is when I tested the code, I was in a context where it did throw exceptions. But then, it wouldn't necessarily be called in that context.